std/path.d

```diff
@@ -1725,21 +1725,23 @@ if (Ranges.length >= 2 &&
     Returns: The assembled path.
 */
 immutable(C)[] buildNormalizedPath(C)(const(C[])[] paths...)
-    @trusted pure nothrow
+    @safe pure nothrow
 if (isSomeChar!C)
 {
     import std.array : array;
+    import std.exception : assumeUnique;
 
-    const(C)[] result;
+    const(C)[] chained;
     foreach (path; paths)
     {
-        if (result)
-            result = chainPath(result, path).array;
+        if (chained)
+            chained = chainPath(chained, path).array;
         else
-            result = path;
+            chained = path;
     }
-    result = asNormalizedPath(result).array;
-    return cast(typeof(return)) result;
+    auto result = asNormalizedPath(chained);
+    // .array returns a copy, so we know it is unique
+    return () @trusted { return assumeUnique(result.array); } ();
```

Comment:
Hmm. If `.array` returns a unique array, how come the compiler can't infer it?  I would have thought that it should be inferred as `pure`, and the compiler would have been able to implicitly convert it to immutable without a `@trusted` escape.

Reply:
Pure isn't enough to infer the result is immutable.